frontend/packages/dev-console/src/components/pipelines/modals/common/SecretForm.tsx

```diff
@@ -63,17 +66,17 @@ const SecretForm: React.FC<FormikValues> = ({
   const setValues = (type: SecretType) => {
     if (type === SecretType.dockerconfigjson) {
       setFieldValue(
-        'formData',
+        'secretForm.formData',
         _.mapValues({ '.dockerconfigjson': stringData[type] }, JSON.stringify),
       );
     } else {
-      setFieldValue('formData', stringData[type]);
+      setFieldValue('secretForm.formData', stringData[type]);
     }
   };
 
   const onDataChanged = (value: string) => {
-    setStringData(_.merge(stringData, { [values.type]: value }));
-    setValues(values.type);
+    setStringData(_.merge(stringData, { [secretForm.type]: value }));
```

Comment:
I know this predates you, but `_.merge` [mutates](https://lodash.com/docs/4.17.15#merge) the first value. Thus mutating React useState.

```suggestion
    setStringData(_.merge({}, stringData, { [secretForm.type]: value }));
```